streaming/src/main/scala/org/apache/carbondata/streaming/StreamSinkFactory.scala

```diff
@@ -45,11 +50,41 @@ import org.apache.carbondata.streaming.segment.StreamSegment
  */
 object StreamSinkFactory {
 
+  val LOGGER = LogServiceFactory.getLogService(this.getClass.getCanonicalName)
+
+  val locks = new util.concurrent.ConcurrentHashMap[String, ICarbonLock]()
+
+  def lock(carbonTable: CarbonTable): Unit = {
```

Comment:
I think it is not intuitive to have lock/unlock function in a Factory class. Can you put it in other place

Reply:
ok, I will modify it in the future.
This is a synchronized pr, better to keep same with previous.